mkosi/config.py

```diff
@@ -1948,6 +1924,45 @@ def parse_config(argv: Sequence[str] = ()) -> tuple[MkosiArgs, tuple[MkosiConfig
     settings_lookup_by_dest = {s.dest: s for s in SETTINGS}
     match_lookup = {m.name: m for m in MATCHES}
 
+    @contextlib.contextmanager
+    def parse_new_includes(
+        namespace: argparse.Namespace,
+        defaults: argparse.Namespace,
+    ) -> Iterator[None]:
+        l = len(getattr(namespace, "include", []))
+
+        try:
+            yield
+        finally:
+            for p in getattr(namespace, "include", [])[l:]:
```

Comment:
Ah, now I finally see it. But I don't think I understand it. For any list `a` the expression `a[len(a):]` will be the empty list, no?

Reply:
The list could have been modified after yielding

Reply:
True, but then put some comments around this, so that the intent is clear to only parse the includes here that existed when this function was called.